src/core/raster/qgsrasterlayertemporalproperties.h

```diff
@@ -144,6 +144,90 @@ class CORE_EXPORT QgsRasterLayerTemporalProperties : public QgsMapLayerTemporalP
      */
     QList< int > filteredBandsForTemporalRange( QgsRasterLayer *layer, const QgsDateTimeRange &range ) const;
 
+    /**
+     * Returns the band number from which the temporal values should be taken.
+     *
+     * \note This is only considered when mode() is Qgis::RasterTemporalMode::RepresentsTemporalValues.
+     * \see setTemporalRepresentationBandNumber()
+     * \since QGIS 3.38
+     */
+    int temporalRepresentationBandNumber() const;
```

Comment:
I'd go with just bandNumber/setBandNumber here. We may end up adding other band-specific modes in future, and I wouldn't want a separate band getter/setter for each of them.